### PR TITLE
fontconfig text font loading for *nix

### PR DESCRIPTION
**Pull request by dpogue**
This uses fontconfig to find the closest match font for the requested name (which I think is always "Trebuchet MS Bold" or "Courier New" right now) on Linux for plTextFont

### CODE REVIEW

File: vcpkg.json [hunk 1]
```diff
@@ -6,6 +6,10 @@
       "asio",
       "curl",
       "expat",
+      {
+        "name": "fontconfig",
+        "platform": "linux"
```
Comment (Hoikas) [inline vcpkg.json:11]:
```suggestion
        "platform": "!(osx | windows)"
```

CMake and the code seem to want fontconfig for non-Apple unixy systems, so we should match that with our platform expression.

File: Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp [hunk 2]
```diff
@@ -150,9 +152,37 @@ uint16_t  *plTextFont::IInitFontTexture()
...
+
+    if (result) {
+        FcPatternDestroy(match);
+
+        FT_Done_FreeType(library);
+        return nullptr;
+    }
+
+    char filename[PATH_MAX];
+    strncpy(filename, (char*)(value.u.s), PATH_MAX - 1);
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp:180]:
```suggestion
    strncpy(filename, (char*)(value.u.s), std::size(filename) - 1);
    filename[std::size(filename) - 1] = 0;
```

File: vcpkg.json (hunk 1 above)
Comment (dpogue) [reply vcpkg.json]: This matched what we had for libuuid, but I've updated it as requested.

However, I think the `$<$<PLATFORM_ID:Linux>:Fontconfig::Fontconfig>` check might still only pick this up on Linux specifically?

File: Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp (hunk 2 above)
Comment (dgelessus) [reply Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp]: Could all the copying logic be skipped if you pass `value.u.s` directly into `FT_New_Face`?

Comment (dpogue) [reply Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp]: I feel like I tried that initially and it blew up 😕

**Review by Hoikas — APPROVED** [review]
#YOLO

**Pull request merged**